packages/fuels-core/src/types/wrappers/input.rs

```diff
@@ -36,6 +37,21 @@ impl Input {
             resource,
             code,
             data,
+            gas_used: 0,
+        }
+    }
+
+    pub const fn resource_predicate_with_gas(
```

Comment:
As it is already  a breaking change I would add the `gas_used` directly to `resource_predicate`